packages/joint-core/src/dia/groups/CellGroup.mjs

```diff
@@ -0,0 +1,67 @@
+import { Model, Collection } from '../../mvc/index.mjs';
+
+export class CellGroupCollection extends Collection {
+
+    _prepareModel(attrs, _options) {
+        if (this._isModel(attrs)) {
+            // do not change collection attribute of a cell model
```

Comment:
Why did you omit this line?
```ts
if (!attrs.collection) attrs.collection = this;
```

These are low-level features and changing them must be properly explained in the code.